### PR TITLE
Add Subscribers.Await for getting published value synchronously

### PR DESCRIPTION
**Pull request by ddddxxx**

### CODE REVIEW

File: Sources/CXExtensions/Await.swift [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import CXShim
+import Foundation
+import CoreFoundation
+
+extension Publisher {
+    
+    public func sync() -> Subscribers.Await<Output, Failure> {
```
Comment (ddddxxx) [inline Sources/CXExtensions/Await.swift]: Naming suggestion?

candidate:
- synchronized
- wait
- blocking

File: Sources/CXExtensions/Await.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        public func next() -> Input? {
+            lock.lock()
+            guard case .idle = demandingState else {
+                lock.unlock()
+                fatalError()
+            }
+            switch subscribingState {
+            case .awaitingSubscription:
+                lock.unlock()
+                fatalError()
```
Comment (ddddxxx) [inline Sources/CXExtensions/Await.swift]: should we crash here or just `return nil`?

File: Sources/CXExtensions/Await.swift (hunk 1 above)
Comment (luoxiu) [reply Sources/CXExtensions/Await.swift]: Sorry for the late reply, well, `wait` is clearer maybe?

```swift
// NSCondition
condition.wait()

// Operation
op.wait()

// DispatchGroup
g.wait()
```

File: Sources/CXExtensions/Await.swift (hunk 2 above)
Comment (luoxiu) [reply Sources/CXExtensions/Await.swift]: `return nil` would be more friendly, : d

File: Sources/CXExtensions/Await.swift (hunk 1 above)
Comment (ddddxxx) [reply Sources/CXExtensions/Await.swift]: This operation does not actually "wait" anything until you iterate it.
```swift
let sub = pub.wait() // non blocking
sub.next() // block here
```
`wait` seems like a blocking operation. It's misleading IMO.

File: Sources/CXExtensions/Await.swift (hunk 2 above)
Comment (ddddxxx) [reply Sources/CXExtensions/Await.swift]: I believe it violate the contract of `IteratorProtocol`. Once `nil` is returned, all subsequent calls should also return `nil`.
```swift
let sub = Await<Int, Never>()
let pub = Just(1)
sub.next() // nil
pub.subscribe(sub)
sub.next() // ⚠️ 1
```

File: Sources/CXExtensions/Await.swift (hunk 1 above)
Comment (ddddxxx) [reply Sources/CXExtensions/Await.swift]: I take `Blocking` from [RXBlocking](https://github.com/ReactiveX/RxSwift/tree/master/RxBlocking)

Comment (luoxiu) [reply Sources/CXExtensions/Await.swift]: make sense

**Review by luoxiu — APPROVED** [review]

**Pull request merged**